dronegen/publish.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+const relcliImage = "146628656107.dkr.ecr.us-west-2.amazonaws.com/gravitational/relcli:v1.1.70-beta.3"
+
+func publishReleasePipeline() pipeline {
+	p := newKubePipeline("publish-rlz")
+	p.Environment = map[string]value{
+		"RELCLI_IMAGE": {raw: relcliImage},
+	}
+	p.Trigger = trigger{
+		Event:  triggerRef{Include: []string{"promote"}},
+		Target: triggerRef{Include: []string{"production"}},
+		Repo:   triggerRef{Include: []string{"gravitational/*"}},
+	}
+	p.Steps = []step{
+		{
+			Name:  "Check if commit is tagged",
+			Image: "alpine",
+			Commands: []string{
+				`[ -n ${DRONE_TAG} ] || (echo 'DRONE_TAG is not set. Is the commit tagged?' && exit 1)`,
+			},
+		},
```

Comment:
Given that this pipeline is triggered by promotion it probably needs a step that verifies if the tag is a pre-release or not

Reply:
This is handled internally by relcli, see https://github.com/gravitational/cloud/blob/3d33121bfd62cfec13131320d350b70c5c26d51d/scripts/relcli/internal/cli/auto_publish.go#L26